src/processes/one_photon_compton/perturbative/cross_section.jl

```diff
@@ -0,0 +1,173 @@
+#####
+# Perturbative one-photon Compton scattering
+# Implementation of the cross section interface
+#####
+
+function _incident_flux(
+    proc::Compton, model::PerturbativeQED, in_ps::AbstractVector{T}
+) where {T<:QEDbase.AbstractFourMomentum}
+    return prod(in_ps)
+end
+
+function _matrix_element(
+    proc::Compton,
+    model::PerturbativeQED,
+    in_ps::AbstractVector{T},
+    out_ps::AbstractVector{T},
+) where {T<:QEDbase.AbstractFourMomentum}
+    return _pert_compton_matrix_element(proc, in_ps, out_ps)
+end
+
+"""
+
+!!! note "Convention"
+
+    We average over the initial spins and pols, and sum over final.
+"""
+function _averaging_norm(proc::Compton)
+    normalizations = number_of_spin_pol.(_in_spin_and_pol(proc))
+    return inv(prod(normalizations))
+end
+
+function _all_onshell(
+    proc::Compton, in_ps::AbstractVector{T}, out_ps::AbstractVector{T}
+) where {T<:QEDbase.AbstractFourMomentum}
+    sq_in_moms = getMass2.(in_ps)
+    sq_out_moms = getMass2.(out_ps)
+    sq_in_masses = mass.(incoming_particles(proc)) .^ 2
+    sq_out_masses = mass.(outgoing_particles(proc)) .^ 2
+    return isapprox(sq_in_moms, SVector(sq_in_masses)) &&
+           isapprox(sq_out_moms, SVector(sq_out_masses))
+end
+
+function _is_in_phasespace(
+    proc::Compton,
+    model::PerturbativeQED,
+    in_ps_def::AbstractPhasespaceDefinition,
+    in_ps::AbstractVector{T},
+    out_ps::AbstractVector{T},
+) where {T<:QEDbase.AbstractFourMomentum}
+    return (!isapprox(sum(in_ps), sum(out_ps))) ? false : _all_onshell(proc, in_ps, out_ps)
```

Comment:
Becomes a little easier to read by leaving the negation out and swapping the results.
```suggestion
    return (isapprox(sum(in_ps), sum(out_ps))) ? _all_onshell(proc, in_ps, out_ps) : false
```